subsys/bluetooth/host/hci_core.c

```diff
@@ -4233,6 +4233,13 @@ int bt_disable(void)
 	/* Some functions rely on checking this bitfield */
 	memset(bt_dev.supported_commands, 0x00, sizeof(bt_dev.supported_commands));
 
+	/* Reset IDs and corresponding keys. */
+	bt_dev.id_count = 0;
+#if defined(CONFIG_BT_SMP)
+	bt_dev.le.rl_entries = 0;
+	bt_keys_reset();
```

Comment:
Should `bt_dev.le.rl_entries = 0;` have been part of `bt_keys_reset`?

Reply:
Or should we have a `bt_id_reset` that does this?

Reply:
Good point. I will leave that up to you and @jori-nordic

Reply:
@rugeGerritsen let's merge as-is and follow-up with another one if necessary. `id.c` is "here be dragons" territory.

Reply:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/2520413/0d63d577-9aa1-49c0-a448-255fb70532d4)